### PR TITLE
Fix PL token classification examples

### PR DESCRIPTION
**Pull request by vblagoje**
This PR fixes the following:

- fetches germeval_14 dataset from a new [location](https://drive.google.com/drive/folders/1kC0I2UGl2ltrluI9NqDjaQJGw5iliw_J)  where it has been moved recently (cc @stefan-it)
- correctly implements `def get_dataloader(self, mode: int, batch_size: int, shuffle: bool = False) -> DataLoader:` from BaseTransformer PL parent class (cc @sshleifer )

I have verified both normal and PL training works as expected. Will add tests as we rework examples to use datasets

### CODE REVIEW

**Comment by vblagoje** [comment]
@stefan-it At the bottom of the germeval [page](https://sites.google.com/site/germeval2014ner/data?authuser=0) I found the new location for the datasets (see Downloads section)

**Review by sshleifer — APPROVED** [review]

**Comment by sshleifer** [comment]
would love tests!

**Comment by vblagoje** [comment]
I will start working on them @sshleifer

**Comment by stefan-it** [comment]
I updated the urls a while ago in this PR https://github.com/huggingface/transformers/pull/6571 😅

**Comment by vblagoje** [comment]
> I updated the urls a while ago in this PR #6571 😅

Apologies @stefan-it, I didn't know about it. There was an error in PL version of the training so I thought why not fix the dataset URL as well. I am following you now so I'll know more about your PRs. Will you fix the NLP dataset germeval_14 as well?

**Comment by nateraw** [comment]
@vblagoje Thanks for this!!

please reach out to me on [Lightning's Slack](https://join.slack.com/t/pytorch-lightning/shared_invite/zt-f6bl2l0l-JYMK3tbAgAmGRrlNr00f1A) (username same as here on github) or email me: `nate [at] pytorchlightning.ai`. I'm about to make updates across examples this week and would love to sync up with you on this.
